skcosmo/SparseMethods/SparseKPCA.py

```diff
@@ -120,26 +126,28 @@ def fit_transform(self, X, X_sparse=None, y=None):
         self.fit(X, X_sparse, y)
         self.transform(X)
 
-    def inverse_transform(self, X):
-        """Transform X back to original space.
+    def inverse_transform(self, T):
+        """Transform T back to original space.
 
-        Parameters
-        ----------
-        X : {array-like} of shape (n_samples, n_components)
 
-        Returns
-        -------
-        X_new : ndarray of shape (n_samples, n_features)
+        :param T : matrix for inverse transform
+        :type T: ndarray of shape (n_samples, n_features)
+
+        :return X_new: result of inverse transformation
+        :type X_new: ndarray of shape (n_samples, n_features)
 
         References
         ----------
         "Learning to Find Pre-Images", G BakIr et al, 2004.
         """
+
+        check_is_fitted(self, ["ptx_"])
+
```

Comment:
You should do this _after_ checking `self.fit_inverse_transform`, otherwise the more specific error will not be raised.

Reply:
Ok, changed it.